src/sql/workbench/services/queryEditor/browser/queryEditorService.ts

```diff
@@ -92,16 +92,14 @@ export class QueryEditorService implements IQueryEditorService {
 		if (sqlContent) {
 			m.textEditorModel.setValue(sqlContent);
 		}
-		m.onDidChangeDirty(() => {
-			if (m.isDirty()) {
-				m.setDirty(false);
-			}
-		});
 
 		// Create an EditDataInput for editing
 		const resultsInput: EditDataResultsInput = this._instantiationService.createInstance(EditDataResultsInput, docUri.toString());
 		let editDataInput: EditDataInput = this._instantiationService.createInstance(EditDataInput, docUri, schemaName, tableName, fileInput, sqlContent, resultsInput);
 
+		editDataInput.onDidChangeDirty(() => {
```

Comment:
I'd rather this go in the actual editDataInput file - that keeps the code together rather than spreading it across different places. This is something we want all Edit Data editors to have so no reason to tie it to this specific place (even if this is currently the only place we make this - that might change and relying on us remembering to copy this over is likely to just result in subtle bugs)